site/app/controllers/admin/ReportController.php

```diff
@@ -776,8 +772,56 @@ public function displayGradebook() {
             new WebResponse(
                 ['admin', 'Report'],
                 'showFullGradebook',
-                $grade_file
+                $grade_file,
+                $grade_summaries_last_run
             )
         );
     }
+
+    /**
+     * Generate a custom filename for the downloaded CSV file
+     *
+     * @return string
+     */
+    private function generateCustomFilename() {
```

Comment:
It would be better to use native PHP type annotations instead of PHPDoc comments, because native type annotations can be enforced by the language itself.
```suggestion
    /**
     * Generate a custom filename for the downloaded CSV file
     */
    private function generateCustomFilename(): string {
```